drivers/SmartThings/zigbee-window-treatment/src/hanssem/init.lua

```diff
@@ -0,0 +1,259 @@
+-- JJSmart Window Shade ver 0.1
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+local Zigbee = require "st.zigbee"
+local ZigbeeZcl = require "st.zigbee.zcl"
+local Messages = require "st.zigbee.messages"
+local data_types = require "st.zigbee.data_types"
+local ZigbeeConstants = require "st.zigbee.constants"
+local generic_body = require "st.zigbee.generic_body"
+local Log = require "log"
+local window_preset_defaults = require "st.zigbee.defaults.windowShadePreset_defaults"
+
+
+local TUYA_CLUSTER = 0xEF00
+local DP_TYPE_VALUE = "\x02"
+local DP_TYPE_ENUM = "\x04"
+
+local SeqNum = 0
+
+-------- Send Command Function for Tuya Zigbee device -------------
+-- ZigbeeMessageTx:
+--    Uint16: 0x0000
+--    AddressHeader:
+--        src_addr: 0x0000
+--        src_endpoint: 0x01
+--        dest_addr: 0xDEAD
+--        dest_endpoint: 0x01
+--        profile: 0x0104
+--        cluster: OnOff
+--    ZCLMessageBody:
+--        ZCLHeader:
+--            frame_ctrl: 0x00
+--            seqno: 0x00
+--            ZCLCommandId: 0x00
+--        ReadAttribute:
+--            AttributeId: 0x0000
+
+local function SendCommand(device, DpId, Type, Value)
+	local addrh = Messages.AddressHeader(
+		ZigbeeConstants.HUB.ADDR, 					-- Source Address
+		ZigbeeConstants.HUB.ENDPOINT,				-- Source Endpoint
+		device:get_short_address(),			-- Destination Address
+		device:get_endpoint(TUYA_CLUSTER),	-- Destination Address
+		ZigbeeConstants.HA_PROFILE_ID,				-- Profile Id
+		TUYA_CLUSTER						-- Cluster Id
+	)
+	local zclh = ZigbeeZcl.ZclHeader({cmd = data_types.ZCLCommandId(0x00)})
+	zclh.frame_ctrl:set_cluster_specific()	-- sets this frame control field to be cluster specific
+	-- Make a payload body 
+	SeqNum = (SeqNum + 1) % 65536
+	local strSeqNum = string.pack(">I2", SeqNum)  -- Pack the Sequence number to 2 bytes unsigned integer type with big endian.
+	local LenOfValue = string.pack(">I2",string.len(Value))  -- Pack length of Value to 2 bytes unsigned integer type wiht big endian.
+	local PayloadBody = generic_body.GenericBody(strSeqNum .. DpId .. Type .. LenOfValue .. Value)
+	local MsgBody = ZigbeeZcl.ZclMessageBody({zcl_header = zclh, zcl_body = PayloadBody})
+	local TxMsg = Messages.ZigbeeMessageTx({address_header = addrh, body = MsgBody})
+	Log.debug(string.format(" ###########>>>>>>>>>>>> Send DpId : %d   Value : %d", string.byte(DpId,1), string.unpack(">I"..string.len(Value),Value)))
```

Comment:
this logging should be removed for the final version

Reply:
All logging functions have been removed.